src/data/energy.ts

```diff
@@ -721,19 +723,19 @@ export const getEnergyDataCollection = (
     collection.prefs = undefined;
   };
   collection.setPeriod = (newStart: Date, newEnd?: Date) => {
+    if (collection._updatePeriodTimeout) {
+      clearTimeout(collection._updatePeriodTimeout);
+      collection._updatePeriodTimeout = undefined;
+    }
     collection.start = newStart;
     collection.end = newEnd;
     if (
       collection.start.getTime() ===
         calcDate(new Date(), startOfDay, hass.locale, hass.config).getTime() &&
       collection.end?.getTime() ===
-        calcDate(new Date(), endOfDay, hass.locale, hass.config).getTime() &&
-      !collection._updatePeriodTimeout
+        calcDate(new Date(), endOfDay, hass.locale, hass.config).getTime()
     ) {
       scheduleUpdatePeriod();
-    } else if (collection._updatePeriodTimeout) {
-      clearTimeout(collection._updatePeriodTimeout);
-      collection._updatePeriodTimeout = undefined;
     }
```

Comment:
The logic in this function was slightly off, and anytime you set the period, if a timeout was set, it would just cancel that timeout and not replace it, even if we should have a new timeout. 

So if you were set to today's date (timeout is scheduled), and you accidentally clicked the "Today" button, it would _cancel_ the timeout, but not replace it. If you click the button again, it would setup a new timeout. If you click the button again, it would cancel the timeout.

Instead just cancel the pre-existing timeout (if it exists), and then always schedule a new timeout if the date matches.